src/com/goide/inspections/GoStructInitializationInspection.java

```diff
@@ -32,17 +41,87 @@ protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNul
       public void visitCompositeLit(@NotNull GoCompositeLit o) {
         super.visitCompositeLit(o);
 
-        if (PsiTreeUtil.getParentOfType(o, GoReturnStatement.class, GoShortVarDeclaration.class, GoAssignmentStatement.class) == null) return;
+        if (PsiTreeUtil.getParentOfType(o, GoReturnStatement.class, GoShortVarDeclaration.class, GoAssignmentStatement.class) == null) {
+          return;
+        }
+
+        if (!(o.getContainingFile() instanceof GoFile)) return;
+        String myPackage = ((GoFile)o.getContainingFile()).getPackageName();
+
+        List<GoFieldDeclaration> fields;
+        GoType type = o.getType();
+        if (type instanceof GoStructType) {
+          if (!canProcess(myPackage, (GoStructType)type)) return;
+          fields = ((GoStructType)type).getFieldDeclarationList();
+          if (hasAnonymousStructField(fields)) return;
+          processLiteralValue(holder, fields, o.getLiteralValue());
+          return;
+        }
 
         GoTypeReferenceExpression ref = o.getTypeReferenceExpression();
-        if (o.getType() instanceof GoStructType || ref != null && ref.resolveType() instanceof GoSpecType) {
+        if (ref == null || ref.resolveType() == null) return;
+        GoType refType = ref.resolveType().getUnderlyingType();
+        if (refType instanceof GoStructType) {
+          if (!canProcess(myPackage, (GoStructType)refType)) return;
+          fields = ((GoStructType)refType).getFieldDeclarationList();
+          if (hasAnonymousStructField(fields)) return;
+          processLiteralValue(holder, fields, o.getLiteralValue());
```

Comment:
Hmm, I've replaced all 3 if-statements with the following code:

```
GoType refType = GoPsiImplUtil.getLiteralType(o);
        if (refType instanceof GoStructType) {
          processStructType((GoStructType)refType, o, holder);
        }
```

and all tests are passed